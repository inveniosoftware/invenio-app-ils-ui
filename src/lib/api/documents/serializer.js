import { recordResponseSerializer } from '@api/utils';

const DocumentSerializers = {
  responseSerializer: function(hit) {
    return recordResponseSerializer(hit);
  },
};

export const documentSerializer = {
  fromJSON: DocumentSerializers.responseSerializer,
};
