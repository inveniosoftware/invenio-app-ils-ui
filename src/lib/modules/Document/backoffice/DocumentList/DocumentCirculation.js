import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Icon, List } from 'semantic-ui-react';

export default class DocumentCirculation extends Component {
  renderOverbookIcon = (isOverbooked) => {
    const name = isOverbooked ? 'warning sign' : 'minus';
    const color = isOverbooked ? 'red' : 'green';
    return <Icon name={name} color={color} size="small" />;
  };

  render() {
    const { document } = this.props;
    const circulation = document.metadata.circulation;
    return (
      <List verticalAlign="middle" className="document-circulation">
        <List.Item>
          <List.Content floated="right">
            <strong>{document.metadata.items.total}</strong>
          </List.Content>
          <List.Content>total physical copies</List.Content>
        </List.Item>
        <List.Item>
          <List.Content floated="right">
            <strong>{circulation.available_items_for_loan_count}</strong>
          </List.Content>
          <List.Content>available copies for loan</List.Content>
        </List.Item>
        <List.Item>
          <List.Content floated="right">
            <strong>{circulation.active_loans_count}</strong>
          </List.Content>
          <List.Content>active loans</List.Content>
        </List.Item>
        <List.Item>
          <List.Content floated="right">
            <strong>{circulation.pending_loans_count}</strong>
          </List.Content>
          <List.Content>loan requests</List.Content>
        </List.Item>
        <List.Item>
          <List.Content floated="right">
            {this.renderOverbookIcon(circulation.overbooked)}
          </List.Content>
          <List.Content>overbooked</List.Content>
        </List.Item>
      </List>
    );
  }
}

DocumentCirculation.propTypes = {
  document: PropTypes.object.isRequired,
};
