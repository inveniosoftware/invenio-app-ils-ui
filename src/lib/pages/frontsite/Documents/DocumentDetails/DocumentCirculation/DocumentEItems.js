import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Divider, Header } from 'semantic-ui-react';
import { invenioConfig } from '@config';
import { DocumentLinks } from '@modules/Document/DocumentLinks';

export class DocumentEItems extends Component {
  showAll = () => {
    const { showTab } = this.props;
    showTab(5);
    const element = document.getElementById('document-metadata-section');
    element.scrollIntoView({ behavior: 'smooth' });
  };

  render() {
    const { eitems } = this.props;

    return eitems.total > 0 ? (
      <>
        <Header as="h3">Access online</Header>
        <DocumentLinks
          eitems={eitems}
          showMaxLinks={invenioConfig.DOCUMENTS.frontsiteMaxLinks}
          onShowAll={this.showAll}
        />
        <Divider horizontal>Or</Divider>
      </>
    ) : null;
  }
}

DocumentEItems.propTypes = {
  eitems: PropTypes.object,
  showTab: PropTypes.func.isRequired,
};

DocumentEItems.defaultProps = {
  eitems: {},
};
