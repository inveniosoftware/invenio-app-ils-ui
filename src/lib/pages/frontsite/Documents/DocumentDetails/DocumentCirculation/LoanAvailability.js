import { DateTime } from 'luxon';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { List, Popup } from 'semantic-ui-react';

export class LoanAvailability extends Component {
  render() {
    const { circulation } = this.props;
    if (circulation.available_items_for_loan_count > 0) {
      return (
        <List.Item>
          <Popup
            content="Calculated based on current library stock"
            trigger={<List.Icon name="info circle" />}
          />
          <List.Content
            className={
              circulation.available_items_for_loan_count > 0
                ? 'text-success'
                : 'text-danger'
            }
          >
            Available for loan <span className="success">now</span>
          </List.Content>
        </List.Item>
      );
    } else if (circulation.next_available_date) {
      return (
        <List.Item>
          <List.Icon name="info circle" />
          <List.Content>
            Available for loan from{' '}
            <b>
              {DateTime.fromISO(circulation.next_available_date).toLocaleString(
                {
                  month: 'long',
                  day: 'numeric',
                }
              )}
            </b>
          </List.Content>
        </List.Item>
      );
    } else {
      return (
        <List.Item>
          <List.Icon name="info" />
          <List.Content>
            Long waiting queue. Contact the library for more information.
          </List.Content>
        </List.Item>
      );
    }
  }
}

LoanAvailability.propTypes = {
  circulation: PropTypes.object.isRequired,
};
