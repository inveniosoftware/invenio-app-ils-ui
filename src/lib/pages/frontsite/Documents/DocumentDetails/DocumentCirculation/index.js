import { connect } from 'react-redux';
import DocumentCirculationComponent from './DocumentCirculation';
import { showTab } from '../actions';

const mapStateToProps = (state, ownProps) => ({
  isLoading: ownProps.isLoading,
  loanRequestIsLoading: state.loanRequestForm.isLoading,
});

const mapDispatchToProps = (dispatch) => ({
  showTab: (activeIndex) => dispatch(showTab(activeIndex)),
});

export const DocumentCirculation = connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentCirculationComponent);
