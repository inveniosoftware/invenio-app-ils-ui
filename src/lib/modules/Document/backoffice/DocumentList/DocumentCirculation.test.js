import * as testData from '@testData/documents.json';
import { shallow } from 'enzyme';
import React from 'react';
import DocumentCirculation from './DocumentCirculation';

describe('DocumentCirculation tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const document = {
    metadata: {
      ...testData[0],
      items: { total: 2 },
      circulation: { available_items_for_loan_count: 2 },
    },
  };

  it('should load the DocumentCirculation component', () => {
    const component = shallow(<DocumentCirculation document={document} />);
    expect(component).toMatchSnapshot();
  });
});
