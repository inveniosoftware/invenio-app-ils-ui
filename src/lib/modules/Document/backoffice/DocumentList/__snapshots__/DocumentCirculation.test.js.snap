// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCirculation tests should load the DocumentCirculation component 1`] = `
<List
  className="document-circulation"
  verticalAlign="middle"
>
  <ListItem>
    <ListContent
      floated="right"
    >
      <strong>
        2
      </strong>
    </ListContent>
    <ListContent>
      total physical copies
    </ListContent>
  </ListItem>
  <ListItem>
    <ListContent
      floated="right"
    >
      <strong>
        2
      </strong>
    </ListContent>
    <ListContent>
      available copies for loan
    </ListContent>
  </ListItem>
  <ListItem>
    <ListContent
      floated="right"
    >
      <strong />
    </ListContent>
    <ListContent>
      active loans
    </ListContent>
  </ListItem>
  <ListItem>
    <ListContent
      floated="right"
    >
      <strong />
    </ListContent>
    <ListContent>
      loan requests
    </ListContent>
  </ListItem>
  <ListItem>
    <ListContent
      floated="right"
    >
      <Icon
        as="i"
        color="green"
        name="minus"
        size="small"
      />
    </ListContent>
    <ListContent>
      overbooked
    </ListContent>
  </ListItem>
</List>
`;
